src/cms/config/functions/slugValidator.js

```diff
@@ -22,6 +22,12 @@ module.exports = {
       throw strapi.errors.badRequest('Do not add slash to the beginning of slugs.');
     }
   },
+  // a leading dash is not allowed
+  slugNoLeadingDashValidator: function (slug) {
+    if (slug?.slice(0, 1) === '-') {
+      throw strapi.errors.badRequest('Do not add dash to the beginning of slugs.');
+    }
+  },
```

Comment:
you could include a second check
```
if (slug.includes('/-')) {
      throw strapi.errors.badRequest('Do not add dash after a slash.');
}
```

Reply:
You'll need to add slugNoLeadingDashValidator back onto "page" for this work now for 
`/-about/foo`